internal/incident/restore_incidents_test.go

```diff
@@ -0,0 +1,238 @@
+
+	// Since we have been using object.FromEvent() to persist the test objects to the database,
+	// these will be automatically added to the objects cache as well. So clear the cache before
+	// reloading the incidents, otherwise it will panic in object.RestoreObjects().
+	object.ClearCache()
+
+	// The incident loading process may hang due to unknown bugs or semaphore lock waits.
+	// Therefore, give it maximum time of 10s to finish normally, otherwise give up and fail.
+	ctx, cancelFunc := context.WithDeadline(ctx, time.Now().Add(10*time.Second))
+	defer cancelFunc()
+
+	err := LoadOpenIncidents(ctx, db, logger, &config.RuntimeConfig{})
+	require.NoError(t, err, "failed to load not recovered incidents")
+
+	incidents := GetCurrentIncidents()
+	assert.Len(t, incidents, len(testData), "failed to load all active incidents")
+
+	for _, current := range incidents {
+		i := testData[current.ObjectID.String()]
+		assert.NotNilf(t, i, "found mysterious incident that's not part of our test data")
+		assert.NotNil(t, current.Object, "failed to restore incident object")
+
+		if i != nil {
+			assert.Equal(t, i.Id, current.Id, "incidents linked to the same object don't have the same ID")
+			assert.Equal(t, i.Severity, current.Severity, "failed to restore incident severity")
+			assert.Equal(t, i.StartedAt, current.StartedAt, "failed to restore incident started at")
+			assert.Equal(t, i.RecoveredAt, current.RecoveredAt, "failed to restore incident recovered at")
+
+			assert.NotNil(t, current.EscalationState, "incident escalation state map should've initialised")
+			assert.NotNil(t, current.Recipients, "incident recipients map should've initialised")
+			assert.NotNil(t, current.Rules, "incident rules map should've initialised")
+
+			if current.Object != nil {
+				assert.Equal(t, i.Object, current.Object, "failed to fully restore incident")
+			}
+		}
+	}
+}
+
+// makeIncident returns a fully initialised recovered/un-recovered incident.
+//
+// This will firstly create and synchronise a new object from a freshly generated dummy event with distinct
+// tags and name, and ensures that no error is returned, otherwise it will cause the entire test suite to fail.
+// Once the object has been successfully synchronised, an incident is created and synced with the database.
+func makeIncident(ctx context.Context, db *icingadb.DB, t *testing.T, recovered bool) *Incident {
+	ev := &event.Event{
+		Time:     time.Time{},
+		SourceId: 1,
+		Name:     makeRandomString(t),
+		Tags: map[string]string{ // Always generate unique object tags not to produce same sha1!
+			"host":    makeRandomString(t),
+			"service": makeRandomString(t),
+		},
+		ExtraTags: map[string]string{
+			"hostgroup/database-server": "",
+			"servicegroup/webserver":    "",
+		},
+	}
+
+	o, err := object.FromEvent(ctx, db, ev)
+	require.NoError(t, err)
+
+	i := NewIncident(db, o, &config.RuntimeConfig{}, nil)
+	i.StartedAt = types.UnixMilli(time.Unix(time.Now().Add(-2*time.Hour).Unix(), 0))
```

Comment:
Why not just use `types.UnixMilli(time.Now().Add(-2*time.Hour))` here?

Reply:
`types.Scan()` uses `utils#FromUnixMilli()`, which in turn uses `time.Unix()`, and `time.Now()` is just a local time, thus the loaded incidents won't have the same timestamp.

Reply:
What exactly is the difference? Is it timezone information and/or sub-second precision? It might be possible to achieve the same by calling [`UTC()`](https://pkg.go.dev/time#Time.UTC) and/or [`Truncate(time.Second)`](https://pkg.go.dev/time#Time.Truncate) to achieve the same in a more obvious way.

Reply:
```go
expected: types.UnixMilli{wall:0xc1805b940a3e40e0, ext:-7199881851166, loc:(*time.Location)(0x100e1ffa0)}
actual  : types.UnixMilli{wall:0xa3140c3, ext:63849033168, loc:(*time.Location)(0x100e1ffa0)}
```

Reply:
`Truncate(time.Second)` fixes the tests.